core/src/autogluon/core/trainer/abstract_trainer.py

```diff
@@ -1134,6 +1134,15 @@ def save(self):
         if self.low_memory:
             self.models = models
 
+    def compile_models(self, **kwargs):
+        models_to_compile = list(self.models.keys())
+        if len(models_to_compile) == 0:
+            raise RuntimeError("No valid persisted model to be compiled. "
+                               "Make sure make a call to `persist_models` before trying to compile the models.")
+        for model_name in models_to_compile:
+            self.models[model_name].compile(compiler_configs=kwargs)
```

Comment:
```
for model_name in models:
    model = self.load_model(model_name)
    # Check if already compiled or if can't compile due to missing dependencies or if model hasn't implemented compiling
    if model.can_compile(kwargs):
         model.compile(compiler_configs=kwargs)
         model.save()
```

Reply:
1) Added `can_compile` method, 2) added the comment after loading the model.